src/jinja2/runtime.py

```diff
@@ -982,10 +982,15 @@ class ChainableUndefined(Undefined):
     def __html__(self) -> str:
         return str(self)
 
-    def __getattr__(self, _: str) -> "ChainableUndefined":
+    def __getattr__(self, name: str) -> "ChainableUndefined":
+        # unimplemented dunder methods returning Undefined break core Python protocols
+        if name[:2] == "__":
+            raise AttributeError(name)
+
         return self
 
-    __getitem__ = __getattr__  # type: ignore
+    def __getitem__(self, _: str) -> "ChainableUndefined":  # type: ignore
```

Comment:
Don't use `_` as an argument name. Use specific ignores, not a bare ignore.

Reply:
lol, that's what PyCharm coughed up for the override signature- I assumed the base class did the same, but apparently it was just confused by the dynamic `_fail_with_undefined_error` aliasing. 

The base class is already doing the same blanket `type: ignore` on that method due to the aforementioned dynamic aliasing- I detest overly-broad ignores as well, so happy to figure out a more targeted ignore (and of course use a proper arg name).

Reply:
Remove the ignore, run mypy, add an ignore for the actual finding. Yeah, there's still a bunch of bare ignores from when we first added typing, but I don't want to persist that going forward.